### PR TITLE
Add option to use resolvable and non-resolvable private address.

### PR DESCRIPTION
**Pull request by asukiaaa**
To use RPA or NRPA without editing code of library.
Reference: https://github.com/h2zero/NimBLE-Arduino/issues/172

### CODE REVIEW

**Comment by h2zero** [comment]
This is a good beginning, however like I said in your issue thread this API should be in the NimBLEDevice class and all the other places this value is used should call a method on it to get the value.

Here are the other places:
https://github.com/h2zero/NimBLE-Arduino/blob/bc52eac5618c45bcfc00b1004bc25b3b1709f0ae/src/NimBLEClient.cpp#L197
https://github.com/h2zero/NimBLE-Arduino/blob/5cb454bfce3bf753e26d00817b7bf66cc0de447a/src/NimBLEAdvertising.cpp#L476

Also, in the call to set the address type we should make the appropriate call to `ble_hs_pvcy_rpa_config();` Not a whole lot to change and I would be more than happy to merge it with those changes.

**Comment by asukiaaa** [comment]
@h2zero 
Thank you for showing the specific code.
Maybe I got it.
I update code to handle m_own_addr through NimBLEDevice.
How about this?

File: src/NimBLEDevice.cpp [hunk 1]
```diff
@@ -705,6 +707,36 @@ void NimBLEDevice::setSecurityCallbacks(NimBLESecurityCallbacks* callbacks) {
...
+ * The available bits are defined as:
+ * * 0x00: BLE_OWN_ADDR_PUBLIC
+ * * 0x01: BLE_OWN_ADDR_RANDOM
+ * * 0x02: BLE_OWN_ADDR_RPA_PUBLIC_DEFAULT
+ * * 0x03: BLE_OWN_ADDR_RPA_RANDOM_DEFAULT
+ */
+void NimBLEDevice::setOwnAddrType(uint8_t own_addr_type, bool useNRPA) {
+    m_own_addr_type = own_addr_type;
+    if (own_addr_type != BLE_OWN_ADDR_PUBLIC) {
+        setSecurityInitKey(BLE_SM_PAIR_KEY_DIST_ID);
```
Comment (h2zero) [inline src/NimBLEDevice.cpp]: This needs to be `setSecurityInitKey(BLE_SM_PAIR_KEY_DIST_ENC | BLE_SM_PAIR_KEY_DIST_ID);` or we won't be able to encrypt connections.

File: src/NimBLEDevice.cpp [hunk 2]
```diff
@@ -705,6 +707,36 @@ void NimBLEDevice::setSecurityCallbacks(NimBLESecurityCallbacks* callbacks) {
...
+        setSecurityRespKey(BLE_SM_PAIR_KEY_DIST_ID);
```
Comment (h2zero) [inline src/NimBLEDevice.cpp]: This line can be removed, as the default setting in `init()` is ` BLE_SM_PAIR_KEY_DIST_ENC | BLE_SM_PAIR_KEY_DIST_ID`

**Comment by h2zero** [comment]
Looks great! thank you for this. just a couple minor comments to address.

**Comment by asukiaaa** [comment]
Thank you for checking.
I call `setSecurityInitKey(BLE_SM_PAIR_KEY_DIST_ENC | BLE_SM_PAIR_KEY_DIST_ID);` in `setOwnAddrType`.

**Comment by h2zero** [comment]
Looks good to me, thank you!

**Comment by h2zero** [comment]
Just noticed this before merging:

https://github.com/h2zero/NimBLE-Arduino/blob/5cb454bfce3bf753e26d00817b7bf66cc0de447a/src/NimBLEAdvertising.cpp#L481

That line needs to have the address type change as well, my mistake, it should have been mentioned earlier.

**Comment by asukiaaa** [comment]
Thank you for checking carefully.
I updated.

**Pull request merged**